drivers/src/mailbox.rs

```diff
@@ -394,15 +431,60 @@ impl MailboxRecvTxn<'_> {
         Ok(())
     }
 
+    /// Sends `data.len()` bytes to the mailbox FIFO
+    /// Transitions from Execute --> RdyForData
     ///
-    /// Transitions from Execute --> Idle
+    /// # Arguments
     ///
-    pub fn complete(&mut self, success: bool) -> CaliptraResult<()> {
+    /// * `data` - data buffer.
+    ///
+    /// # Returns
+    ///
+    /// Status of Operation
+    ///
+    pub fn copy_response(&mut self, data: &[u8]) -> CaliptraResult<()> {
         if self.state != MailboxOpState::Execute {
             return Err(CaliptraError::DRIVER_MAILBOX_INVALID_STATE);
         }
+
+        self.state = MailboxOpState::RdyForDlen;
+        // Set dlen
+        self.write_dlen(data.len() as u32)?;
+
+        self.state = MailboxOpState::RdyForData;
+        // Copy the data
+        self.enqueue(data)
+    }
+
+    /// Sends `data.len()` bytes to the mailbox FIFO.
+    /// Transitions from Execute --> Idle (releases the lock)
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - data buffer.
+    ///
+    /// # Returns
+    ///
+    /// Status of Operation
+    ///
+    pub fn send_response(&mut self, data: &[u8]) -> CaliptraResult<()> {
+        self.copy_response(data)?;
+        self.complete(true)
+    }
+
+    ///
+    /// Transitions from Execute or RdyForData-> Idle
+    ///
+    pub fn complete(&mut self, success: bool) -> CaliptraResult<()> {
```

Comment:
This can be `mut self` instead of `&mut self`.

Reply:
No problem